src/parse.rs

```diff
@@ -350,59 +351,35 @@ pub fn parse_ids(
     }
 }
 
+/// Parse a single unsolicited response from IDLE responses.
+pub fn parse_idle(lines: &[u8]) -> (&[u8], Option<Result<UnsolicitedResponse>>) {
+    match imap_proto::parser::parse_response(lines) {
+        Ok((rest, response)) => match UnsolicitedResponse::try_from(response) {
```

Comment:
Are there any responses that can come during IDLE that cannot come as unsolicited responses otherwise?

Reply:
Strictly speaking, no - nothing in the IDLE spec that I have seen says that there are special Responses that only come on an IDLE connection. It just says that it will send unilateral responses and doesn't itemize what those responses will be outside of some examples which are not meant to be complete. As near as I can tell, IDLE is just saying that the client is signing up for arbitrary unilateral responses of any kind. In practice, the practical range of responses that will come on an Idle channel is smaller than the set of responses that will come otherwise (at least as far as I have seen).